### PR TITLE
[PM-1390] X-Plat/SDK Style SQL Project

### PR DESCRIPTION
**Pull request by justindbaur**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [x] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Uses the new, in preview, SDK style SQL Database Project that is cross-platform. It uses glob file matching so that we don't have to add individual files to the project. As long as it follows the [`**/*.sql`](https://github.com/microsoft/DacFx/blob/13d97d6ff36dbfc5959b367aa5389caa58cb4f05/src/Microsoft.Build.Sql/sdk/Sdk.props#L61) pattern it will automatically get grabbed. This also makes it so that if our Mac and Linux developers run `dotnet build` from root it will have 0 errors! (unless I just added them)

> WARNING: This is in preview so we might want to hold back but since we don't actually use this package to generate dacpac files I think it is lower risk. 

Read More:
- [Introduction Article](https://learn.microsoft.com/en-us/sql/azure-data-studio/extensions/sql-database-project-extension-sdk-style-projects?view=sql-server-ver16)
- [Source Code](https://github.com/microsoft/DacFx/tree/13d97d6ff36dbfc5959b367aa5389caa58cb4f05/src/Microsoft.Build.Sql)

## Code changes

* **Sql.sqlproj:** Migrated it to SDK style. 
* **src/Sql/dbo/Stored Procedures/OrganizationDomain_Update.sql:** Fixed syntax error it complained about
* **.github/workflows/database.yml:** Change build to work all in a single step in a ubuntu runner

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: src/Sql/packages.lock.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{
+  "version": 1,
+  "dependencies": {
+    ".NETFramework,Version=v4.6": {
```
Comment (Hinton) [inline src/Sql/packages.lock.json:4]: Shouldn't this be targeting the .netstandard?

File: src/Sql/dbo/Stored Procedures/OrganizationDomain_Update.sql [hunk 2]
```diff
@@ -20,9 +20,9 @@ BEGIN
         [DomainName] = @DomainName,
         [CreationDate] = @CreationDate,
         [VerifiedDate] = @VerifiedDate,
-        [LastCheckedDate] = @LastCheckedDate
+        [LastCheckedDate] = @LastCheckedDate,
```
Comment (Hinton) [inline src/Sql/dbo/Stored Procedures/OrganizationDomain_Update.sql:23]: Seems odd that these were let through before?

**Review by Hinton — COMMENTED** [review]
I think it's worth noting that this removes support for running sql proj in visual studio on windows. On the flip side this should make it possible to compare the schema using azure data studio?

**Comment by justindbaur** [comment]
> I think it's worth noting that this removes support for running sql proj in visual studio on windows. On the flip side this should make it possible to compare the schema using azure data studio?

It works fine in Visual Studio 2022 Preview for me, maybe the stable version doesn't have support quite yet. But yes schema compare should be supported with an extension in Azure Data Studio, and with command line tools.

File: src/Sql/dbo/Stored Procedures/OrganizationDomain_Update.sql (hunk 2 above)
Comment (justindbaur) [reply src/Sql/dbo/Stored Procedures/OrganizationDomain_Update.sql:23]: Yeah it's a bit odd, possibly a slightly different rule set. There are also plenty of warnings you get when building it but this was the only error.

File: .github/workflows/database.yml [hunk 3]
```diff
@@ -20,53 +20,29 @@ on:
...
       - name: Build DACPAC
-        run: msbuild src/Sql/Sql.sqlproj /p:Configuration=Release /verbosity:minimal
+        run: dotnet build src/Sql --configuration Release --verbosity minimal --output .
         shell: pwsh
 
-      - name: Upload DACPAC
-        uses: actions/upload-artifact@6673cd052c4cd6fcf4b4e6e60ea986c889389535
-        with:
-          name: sql.dacpac
-          path: src/Sql/bin/Release/Sql.dacpac
```
Comment (Hinton) [inline .github/workflows/database.yml:55]: Would be nice to still publish the dacpac.

File: src/Sql/Sql.sqlproj [hunk 4]
```diff
@@ -1,484 +1,13 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <Project DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003" ToolsVersion="4.0">
```
Comment (Hinton) [inline src/Sql/Sql.sqlproj:2]: The docs seems to suggest this is no longer necessary.

```suggestion
<Project DefaultTargets="Build">
```

**Review by Hinton — DISMISSED** [review]

**Comment by justindbaur** [comment]
@Hinton Had to fix a merge conflict and add pinned version numbers.

**Review by Hinton — APPROVED** [review]

**Pull request merged**